src/hexen/sv_save.c

```diff
@@ -2043,29 +2042,33 @@ void SV_LoadGame(int slot)
     M_snprintf(fileName, sizeof(fileName), "%shex6.hxs", SavePath);
 
     // Load the file
-    M_ReadFile(fileName, &SaveBuffer);
+    SV_OpenRead(fileName);
 
     // Set the save pointer and skip the description field
-    SavePtr.b = SaveBuffer + HXS_DESCRIPTION_LENGTH;
+    fseek(SavingFP, HXS_DESCRIPTION_LENGTH, SEEK_CUR);
 
     // Check the version text
-    if (strcmp((char *) SavePtr.b, HXS_VERSION_TEXT))
+
+    for (i = 0; i < 16; ++i)
+    {
+        version_text[i] = SV_ReadByte();
+    }
+    if (strcmp(version_text, HXS_VERSION_TEXT) != 0)
```

Comment:
If version_text is not NUL-terminated this will cause a crash. Please use strncmp.